spyder_kernels/utils/misc.py

```diff
@@ -10,6 +10,64 @@
 
 import re
 
+
+# Mapping of inline figure formats
+INLINE_FIGURE_FORMATS = {
+    '0': 'png',
+    '1': 'svg'
+}
+
+MPL_BACKENDS_TO_SPYDER = {
+    'module://ipykernel.pylab.backend_inline': 0,
+    'Qt5Agg': 2,
+    'Qt4Agg': 3,
+    'MacOSX': 4,
+    'GTK3Agg': 5,
+    'GTKAgg': 6,
+    'WX': 7,
+    'TkAgg': 8
+}
+
+
+def is_module_installed(module_name):
+    """
+    Simpler version of spyder.utils.programs.is_module_installed.
+    """
+    try:
+        __import__(module_name)
+        return True
+    except Exception:
+        # Module is not installed
+        return False
+
+
+def automatic_backend():
+    """Get Matplolib automatic backend option."""
+    if is_module_installed('PyQt5'):
+        auto_backend = 'qt5'
+    elif is_module_installed('PyQt4'):
+        auto_backend = 'qt4'
+    elif is_module_installed('_tkinter'):
+        auto_backend = 'tk'
+    else:
+        auto_backend = 'inline'
+    return auto_backend
+
+
+# Mapping of Spyder options to backends
+MPL_BACKENDS_FROM_SPYDER = {
```

Comment:
Please move this constant to be below `MPL_BACKENDS_TO_SPYDER`.

One a side note, it's usually a good practice to declare all constants at the beginning of the file.

Reply:
That's not possible since the `automatic` constant value is calculated with the functions above:

```python
    '1': automatic_backend(),
```

That's why the constant is not below the `MPL_BACKENDS_TO_SPYDER` constant and instead the order of things is:

* `MPL_BACKENDS_TO_SPYDER`
* `is_module_installed` (used by `automatic_backend`)
* `automatic_backend` (used when defining `MPL_BACKENDS_FROM_SPYDER`)
* `MPL_BACKENDS_FROM_SPYDER`

Reply:
Ok, thanks for the claritication @dalthviz. No problem then.